plugins/secretstores/http/aes.go

```diff
@@ -0,0 +1,167 @@
+	}
+	a.mode = mode
+
+	// Setup the trimming function to revert padding
+	switch padding {
+	case "", "none":
+		// identity, no padding
+		a.trim = func(in []byte) ([]byte, error) { return in, nil }
+	case "pkcs#5", "pkcs#7":
+		a.trim = PKCS5or7Trimming
+	default:
+		return fmt.Errorf("unsupported padding %q", padding)
+	}
+
+	// Generate the key using password-based-keys
+	if a.Key.Empty() {
+		if a.Passwd.Empty() {
+			return errors.New("either key or password has to be specified")
+		}
+		if a.Salt.Empty() || a.Iterations == 0 {
+			return errors.New("salt and iterations required for password-based-keys")
+		}
+
+		key, iv, err := a.KDFConfig.NewKey(keylen)
+		if err != nil {
+			return fmt.Errorf("generating key failed: %w", err)
+		}
+		a.Key.Destroy()
+		a.Key = key
+
+		if a.Vec.Empty() && !iv.Empty() {
+			a.Vec.Destroy()
+			a.Vec = iv
+		}
+	} else {
+		encodedKey, err := a.Key.Get()
+		if err != nil {
+			return fmt.Errorf("getting key failed: %w", err)
+		}
+		key := make([]byte, hex.DecodedLen(len(encodedKey)))
+		if _, err := hex.Decode(key, encodedKey); err != nil {
+			config.ReleaseSecret(encodedKey)
+			return fmt.Errorf("decoding key failed: %w", err)
+		}
+		config.ReleaseSecret(encodedKey)
+
+		if len(key) != keylen {
+			return fmt.Errorf("key length (%d bit) does not match cipher (%d bit)", len(key)*8, keylen*8)
+		}
+	}
+
+	if a.Vec.Empty() {
+		return errors.New("'init_vector' has to be specified or derived from password")
+	}
+
+	encodedIV, err := a.Vec.Get()
+	if err != nil {
+		return fmt.Errorf("getting IV failed: %w", err)
+	}
+	ivlen := len(encodedIV)
+	config.ReleaseSecret(encodedIV)
+	if ivlen != 2*aes.BlockSize {
+		return errors.New("init vector size must match block size")
+	}
+
+	return nil
+}
+
+func (a *AesEncryptor) Decrypt(data []byte) ([]byte, error) {
+	if len(data)%aes.BlockSize != 0 {
+		return nil, fmt.Errorf("invalid data size %d", len(data))
+	}
+	if a.mode != "cbc" {
+		return nil, fmt.Errorf("unsupported cipher mode %q", a.mode)
+	}
+
+	// Setup the cipher and return the decoded data
+	encodedKey, err := a.Key.Get()
+	if err != nil {
+		return nil, fmt.Errorf("getting key failed: %w", err)
+	}
+	key := make([]byte, hex.DecodedLen(len(encodedKey)))
+	if _, err := hex.Decode(key, encodedKey); err != nil {
+		config.ReleaseSecret(encodedKey)
+		return nil, fmt.Errorf("decoding key failed: %w", err)
+	}
+	config.ReleaseSecret(encodedKey)
+
+	// Setup AES
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, fmt.Errorf("creating AES cipher failed: %w", err)
+	}
+
+	// Setup the block/stream cipher and decode the data
+	encodedIV, err := a.Vec.Get()
+	if err != nil {
+		return nil, fmt.Errorf("getting initialization-vector failed: %w", err)
+	}
+	iv := make([]byte, hex.DecodedLen(len(encodedIV)))
+	if _, err := hex.Decode(iv, encodedIV); err != nil {
+		config.ReleaseSecret(encodedIV)
+		return nil, fmt.Errorf("decoding init vector failed: %w", err)
+	}
```

Comment:
I made the comment before, but it was resolved: I think we should 'wipe key' here and the previous conditional by assigning it `removed` or something else. Eg:
```
# rewrite key now that we are done using it in case it ends up on the heap, is swapped, etc
key = "removed"
```

Reply:
Wipe the element now in the error case but as the data is passed on to the decrpyter as byte-array I think we cannot wipe `iv`.